src/python/pants/backend/python/goals/coverage_py.py

```diff
@@ -203,10 +207,20 @@ async def create_coverage_config(coverage: CoverageSubsystem) -> CoverageConfig:
             PathGlobs(
                 globs=(coverage.config,),
                 glob_match_error_behavior=GlobMatchErrorBehavior.error,
-                description_of_origin=f"the option `--{coverage.options_scope}-config`",
+                description_of_origin=f"the option `[{coverage.options_scope}].config`",
             ),
         )
         coverage_config.read_string(config_contents[0].content.decode())
+    else:
+        await Get(
```

Comment:
The side effect is jarring but its probably the way to go for now.

Its seems like there might be a seam where you'd use a ConfigFileRequest exposed by an option subsystem as a property and that would force you to get the path value in a way that triggered the logging as a side-effect contained to 1 rule and lifted the knowledge of the subsystem / tool / its config to that 1 spot. If this resonates it may be worth follow up at some point.

Reply:
That's an interesting angle! We right now duplicate lots of code for the config file. I'm going to explore that change as part of this PR while I'm tuned in to the formatter/linter code.

FYI, a future change I want us to add is to detect global config files in $HOME and start warning. I like your proposed factoring because it would be easy to add to that type.